VoiceController.php

```diff
@@ -1,48 +1,34 @@
-public function voice(Request $request){
-    $request->validate([
-        'question_id'=>'required|int|exists:questions,id',
-        'value'=>'required|boolean',
-    ]);
+public function voice(CreateVoiceRequest $request){
+    $question = Question::findOrFail($request->question_id); //find the question or fail
 
-    $question=Question::find($request->post('question_id'));
-    if (!$question)
-        return response()->json([
-            'status'=>404,
-            'message'=>'not found question ..'
-        ]);
-    if ($question->user_id==auth()->id())
-        return response()->json([
-            'status' => 500,
-            'message' => 'The user is not allowed to vote to your question'
-        ]);
+    abort_if($question->user_id === auth()->id(), 403, 'The user is not allowed to vote to your question');
 
     //check if user voted 
-    $voice=Voice::where([
-        ['user_id','=',auth()->id()],
-        ['question_id','=',$request->post('question_id')]
+    $voice = Voice::where([
+        'user_id' => auth()->id(),
+        'question_id' => $request->question_id
     ])->first();
-    if (!is_null($voice)&&$voice->value===$request->post('value')) {
-        return response()->json([
-            'status' => 500,
-            'message' => 'The user is not allowed to vote more than once'
-        ]);
-    }else if (!is_null($voice)&&$voice->value!==$request->post('value')){
+
+    abort_if($voice && $voice->value === $request->value, 403, 'The user is not allowed to vote more than once'); //since the user has a vote already.
```

Comment:
@innoflash not sure I would use `403` code here, it's more like validation for me, so 422. But that's, of course, personal preference.